machine_test.go

```diff
@@ -230,6 +278,39 @@ func (*MachineSerializationSuite) TestNestedParsing(c *gc.C) {
 	c.Assert(machines, jc.DeepEquals, expected)
 }
 
+func (*MachineSerializationSuite) TestNestedParsingWithPriorVersion(c *gc.C) {
```

Comment:
I have to say, I don't find these tests particularly legible. They are probably correct, but it is hard to look at them and clearly understand what they are testing and whether or not they are correct.